### PR TITLE
Readiness Probes Task

### PR DESCRIPTION
**Pull request by pwittrock**

### CODE REVIEW

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+synopsis: "Configure a Pod to start receiving requests only after becoming ready."
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: No synopsis on tasks; we don't surface it anywhere.

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+objects:
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: object_rankings:

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+- readinessProbe
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: - concept: readinessProbe

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+- probe
+concept_rankings:
+- concept: pod
+  rank: 50
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: rank: 2
(Only 1 and 2 are supported right now, these map to "Core" and "Advanced" at the moment)

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+- concept: container
+  rank: 50
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: rank: 2

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+- *ReadinessProbe*s only prevent *Services* from sending traffic to the *Pod*,
+and do not prevent requests from being sent directly to the *Pod* Ip address
+through other means.
+- Each probe will have one of three results:
+  - `Success`: indicates that the container passed the diagnostic.
+  - `Failure`: indicates that the container failed the diagnostic.
+  - `Unknown`: indicates that the diagnostic failed so no action should be taken.
+- Probes can be further customized (e.g. *TimeoutSeconds*, *PeriodSeconds*).  See
+the [probe documentation](http://kubernetes.io/docs/api-reference/v1/definitions/#_v1_probe)
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: URL: /docs/api-reference/v1/... etc

Never include the domain, so it works in staging.

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+It is recommended you are familiar with the following concepts before continuing.
+
+- [Pods](/docs/pod/)
+- [Containers](/docs/container/)
+{% endcapture %}
+
+{% capture step_by_step %}
+#### Option 1: Define an http Get request
+
+Add `pod.containers.readinessProbe.httpGet` with a url and port that returns
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: s/url/URL

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 8]
```diff
@@ -0,0 +1,115 @@
...
+    readinessProbe: # This is the important part
+      exec: # This defines the exec command
+        command: # This is the command + argv executed to check Readiness
+        - cat
+        - /tmp/health
+```
+
+#### Option 3: Define a TCP request
+
+Add `pod.containers.readinessProbe.tcpSocket` with a port that can be opened
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: How about "is opened" and "is not opened" ?

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md [hunk 9]
```diff
@@ -0,0 +1,115 @@
...
+    readinessProbe: # This is the important part
+      tcpSocket: # Connection is made to the Pod IP
+        port: 22
+```
+{% endcapture %}
+
+{% capture options_and_considerations %}
+
+- *ReadinessProbe*s only prevent *Services* from sending traffic to the *Pod*,
+and do not prevent requests from being sent directly to the *Pod* Ip address
```
Comment (johndmulhausen) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: IP address

Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: DOne

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 6 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: Done

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 8 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: DOne

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 7 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: done

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 4 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: Done

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 3 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: Done

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 2 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: DOne

File: docs/tasks/declaring-a-pod-not-ready-for-requests.md (hunk 1 above)
Comment (pwittrock) [inline docs/tasks/declaring-a-pod-not-ready-for-requests.md]: DOne

**Comment by johndmulhausen** [comment]
Nice!

**Pull request merged**